tt_metal/impl/dispatch/kernels/command_queue_producer.hpp

```diff
@@ -172,6 +216,24 @@ void multicore_cb_push_back(uint64_t consumer_noc_encoding, uint32_t consumer_fi
 }
 
 template <uint32_t consumer_cmd_base_addr, uint32_t consumer_data_buffer_size>
+FORCE_INLINE
+void multicore_cb_push_back_2(
+    db_cb_config_t* db_cb_config,
+    const db_cb_config_t* remote_db_cb_config,
+    uint64_t consumer_noc_encoding,
+    uint32_t consumer_fifo_16b_limit,
+    uint32_t num_to_write) {
+    db_cb_config->recv += num_to_write;
+    db_cb_config->wr_ptr += db_cb_config->page_size * num_to_write;
+
+    if (db_cb_config->wr_ptr >= consumer_fifo_16b_limit) {
```

Comment:
put the computation above into a local, save it then compare to save the extra L1 load

Reply:
`db_cb_config` is a local copy already